src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java

```diff
@@ -108,6 +116,61 @@ public void onFailure()
       }
    }
    
+   public void interruptRNoDebug(final InterruptHandler handler) {
+      final int originalDebugType = errorManager_.getErrorHandlerType();
+      
+      if (originalDebugType != ErrorHandlerType.ERRORS_BREAK) {
+         interruptR(handler);
+      }
+      else {
+         errorManager_.setDebugSessionHandlerType(
+            ErrorHandlerType.ERRORS_MESSAGE,
+            new ServerRequestCallback<Void>()
+            {
+               @Override
+               public void onResponseReceived(Void v)
+               {
+                  interruptR(new InterruptHandler() {
+                     @Override
+                     public void onInterruptFinished()
+                     {
+                        errorManager_.setDebugSessionHandlerType(
+                           originalDebugType,
+                           new ServerRequestCallback<Void>()
+                           {
+                              @Override
+                              public void onResponseReceived(Void v)
+                              {
+                                 interruptR(new InterruptHandler() {
```

Comment:
Why do we need to interrupt R a second time here after restoring the handler type?